File: app/src/androidTest/java/name/boyle/chris/sgtpuzzles/TestUtils.java
package name.boyle.chris.sgtpuzzles;

import androidx.test.platform.app.InstrumentationRegistry;

import org.junit.Assume;

public class TestUtils {
    private TestUtils() {}

    public static String getFastlaneDeviceTypeOrSkipTest() {
        final String deviceType = InstrumentationRegistry.getArguments().getString("device_type");
        Assume.assumeNotNull(deviceType);  // only available under fastlane
        return deviceType;
    }
}

File: app/src/androidTest/kotlin/name/boyle/chris/sgtpuzzles/TestUtils.kt
package name.boyle.chris.sgtpuzzles

import androidx.test.platform.app.InstrumentationRegistry
import org.junit.Assume.assumeNotNull

object TestUtils {
    val fastlaneDeviceTypeOrSkipTest: String
        get() {
            val deviceType = InstrumentationRegistry.getArguments().getString("device_type")
            assumeNotNull(deviceType) // only available under fastlane
            return deviceType!!
        }
}